src/app/components/SearchResults/FilterYears.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { initialSearchQuery, searchQueryPropTypes } from '../../stores/InitialState';
+
+import { yearsType } from '../../constants/labels';
+
+const getYearsFilter = (searchQuery) => {
+  const yearsValues = {};
+  Object.keys(yearsType).forEach((yearType) => {
+    const yearValue = searchQuery && searchQuery.filters && searchQuery.filters.find(filter => filter.field === 'years');
+    yearsValues[yearType] = yearValue && yearValue.value[yearType] ? Number(yearValue.value[yearType]) : '';
+  });
+  return yearsValues;
+};
+
+class FilterYears extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = getYearsFilter(props.searchQuery);
+    this.onChangeYear = this.onChangeYear.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState(getYearsFilter(nextProps.searchQuery));
+  }
+
+  onChangeYear(e, yearType) {
+    let val = e.target.value && Number(e.target.value);
+    if (yearType === 'start' && this.state.end && val && val > Number(this.state.end)) {
+      val = this.state.end;
+    }
+    if (yearType === 'end' && this.state.start && val && val < Number(this.state.start)) {
+      val = this.state.start;
```

Comment:
These two options work well but perhaps _too_ well right now. If I enter a year into `start`, say, 2000, and then go to `end` and start typing it immediately picks up that 2 is less than 2000 and autofills 2000 from the `start` field.

I think this should wait until the number of digits in the field being entered is equal to the length of the other field before checking. Otherwise it interrupts the user's flow and forces them to go back and change the date that was entered.

Reply:
Not sure if the length will improve it. Maybe remove all the value changes till we have a design for errors?

Reply:
That works for me, its the way it will work eventually so waiting for the design makes sense. Would that leave the min/max values in place and just remove the automatic value change?

Reply:
Removed the control over max and min. Anyway it shows an error coming from the browser, not designed, for max and min. Maybe that's enough, it looks good